docs/specs/lcb-signals-and-events.md

```diff
@@ -0,0 +1,222 @@
+
+It is an error to evaluate `the signal source` outside a signal handling context.
+
+**Notes:**
+
+1. This allows a single LCB handler to be used to handle multiple signals.  It allows multiple signals with different prototypes to be handled by a single widget's script without overloading.
+
+### Signal connection
+
+When a child widget is added to a composed widget, each declared signal of the child widget is examined.  If the parent widget declares a signal handler with a matching prototype, then it is connected to the signal [1].
+
+When a child widget is removed from a composed widget, all signal handlers in the parent widget that are connected to a signal of the child widget are disconnected.
+
+**Notes:**
+
+1. When these rules are applied to "host widgets", each public script `on` handler in the host widget's script is automatically considered to have been declared as a polymorphic signal handler.
+
+## Events
+
+### Event processing
+
+Events are processed in three phases [1]:
+
+1. **Enter**: before handling an event, perform event-handling setup tasks and determine at point in the event hierarchy the event should be targeted
+
+2. **Dispatch**: do work in response to the event
+
+3. **Leave**: perform any tasks that should be carried out after event handling (for example, tearing down any state set up during event enter)
+
+Before beginning to process an event, the event manager first determines which control should be the **target** of the event (taking into account visibilities, and bounding rectangles) [2].
+
+A list of "interested controls" is built.  This recursively contains every parent of the target.  This is used to determine which objects should have their event handlers run.
+
+**Notes:**
+
+1. The enter phase and leave phase facilitate the implementation of behaviour that wraps normal event dispatch.  For example, consider a composed widget with a border that highlights when the mouse is inside it.  To ensure that highlight changes sequence correctly with with its child widgets, the highlight must be enabled during the `MouseEnter` event's enter phase, and hidden during the `MouseLeave` event's leave phase.
+
+2. How the target of an event is determined may vary a lot on a per-event-type basis (for example, the target for a mouse click will depend on the mouse position, but the target for a keyboard event may depend on the keyboard focus).  For the purposes of this document, assume that it is an arbitrary process.
+
+### Event data
+
+When an event is submitted to the event manager, it has a **event name** and a set of **event data**.  The event data is an array containing all of the captured data relating to the event itself [1].
+
+In an event handler, `the event` evaluates to the event data array [2].
+
+Evaluating `the event` outwith an event handler is an error.
+
+`the event` is immutable.
+
+The `the event target` expression evaluates to the event's target control [3].
+
```

Comment:
As I can't come up with any functional difference between `the event target` and `the target` might I suggest just using the existing syntax?